executor/ddl.go

```diff
@@ -341,6 +371,36 @@ func (e *DDLExec) createSessionTemporaryTable(s *ast.CreateTableStmt) error {
 	return err
 }
 
+func (e *DDLExec) newTemporaryTableFromTableInfo(tbInfo *model.TableInfo) (table.Table, error) {
+	dom := domain.GetDomain(e.ctx)
+	// Local temporary table uses a real table ID.
+	// We could mock a table ID, but the mocked ID might be identical to an existing
```

Comment:
Does the truncate operation require the table ID to change in its semantic?

Reply:
For normal table, it actually changed its table ID. 

https://github.com/pingcap/tidb/blob/c585a76324500dfd1301bc2d348b7f5f4b3e6767/ddl/table.go#L447-L469